### ISSUE TITLE
what is your GAN-loss ？？ 

### ISSUE THREAD

**Issue by Johnson-yue** [lucidrains/stylegan2-pytorch#69]
your adv D_loss is：
https://github.com/lucidrains/stylegan2-pytorch/blob/90c4577c61af1f9442abf2c8b732d8c9b246c4e3/stylegan2_pytorch/stylegan2_pytorch.py#L633

adv G_loss is：
https://github.com/lucidrains/stylegan2-pytorch/blob/90c4577c61af1f9442abf2c8b732d8c9b246c4e3/stylegan2_pytorch/stylegan2_pytorch.py#L667

Hi，I can not understand what is your GAN loss？？
the orignal paper said used non-saturating logistic loss, but you are not .
It seems like HingeLoss, but it different !!

as my opinion HingeLoss([in FUNIT](https://github.com/NVlabs/FUNIT/blob/4cd5f22cc330bbf1804db7d5364e870e412c324a/networks.py#L71-L96))  and other version[ HingeLoss](https://github.com/akanimax/BMSG-GAN/blob/863b1f370c12010a57812fb55e2f1b625e64e5a4/sourcecode/MSG_GAN/Losses.py#L176) is the same  :

divergence should be equal to `(F.relu(1 - real_output) + F.relu(1 + fake_output)).mean()`

Would you tell me, why do you use this format loss,  what is paper you according to ?

**Comment by lucidrains** [comment]
@Johnson-yue Hi Johnson, it doesn't matter. What matters is that the generator and discriminator chooses opposite signs. Whether you assign 0 or 1 as real/fake fake/real can be arbitrarily chosen.

**Comment by Johnson-yue** [comment]
oh， I see， that is good point，thank you
